src/tests.rs

```diff
@@ -455,6 +484,39 @@ fn test_interests() {
     );
 }
 
+fn test_user_span() {
+    let p = g::Pipeline::new();
+
+    let span = tracing::error_span!("pipeline span", pipeline = true);
+    unsafe {
+        p.set_data("tracing-gstreamer:span", span);
```

Comment:
Lets add a top level function that abstracts this away, so that users don’t need to rely on specific key name, what otherwise smells of internal implementation details.

In addition, I think it would make sense to set up a quark on the first use and just keep reusing it, rather than reconstructing it every time these operations are made (data lookups in particular seem like they could be expensive). Something like this I’d imagine:

```rust
#[inline]
fn span_quark() -> &Quark {
    static ELEMENT_SPAN_QUARK: OnceCell<Quark> = OnceCell::new();
    ELEMENT_SPAN_QUARK.get_or_init(|| {
        let gstr = GStr::from_utf8_with_nul(b"tracing-gstreamer:span\0");
        Quark::from_str(gstr)
    })
}

fn attach_span(gobject: &GObject, span: tracing::Span) {
    gobject.set_qdata(span_quark(), span);
}
```

or something?

Reply:
You're right, using a `Quark` is much better here.

I made the object a `IsA<gstreamer::Object>`  as we rely on `GstObject.parent` for the lookup.